module/zfs/zfs_onexit.c

```diff
@@ -126,13 +126,20 @@ zfs_onexit_fd_hold(int fd, minor_t *minorp)
 {
 	file_t *fp;
 	zfs_onexit_t *zo;
+	int error;
 
 	fp = getf(fd);
 	if (fp == NULL)
 		return (SET_ERROR(EBADF));
 
-	*minorp = zfsdev_getminor(fp->f_file);
-	return (zfs_onexit_minor_to_state(*minorp, &zo));
+	error = zfsdev_getminor(fp->f_file, minorp);
+	if (error == 0)
```

Comment:
Since the line comments disappeared as I pushed a revised pull request, I want to copy what I said in response to `[style] The above two if's should just be an if-else block.`:

```
If we do that, then an error in `zfs_zevent_minor_to_state()` would be ignored.
```